lib/prometheus_telemetry/metrics/graphql/request.ex

```diff
@@ -69,12 +79,14 @@ if Enum.any?(Application.loaded_applications(), fn {dep_name, _, _} -> dep_name
       ]
     end
 
-    defp extract_batch_name(%{batch_fun: batch_fun_tuple}) do
-      [module, fnc_name | _] = Tuple.to_list(batch_fun_tuple)
-
-      module = module |> to_string |> String.replace("Elixir.", "")
-
-      %{module: module, function: to_string(fnc_name)}
+    defp parse_name_and_type(%{blueprint: blueprint}) do
+      with {:ok, %Blueprint{input: %{definitions: definitions}}} <- Phase.Parse.run(blueprint),
```

Comment:
Use a cond here since handling all cases!

Reply:
I opted to use regex instead since it would allow for easy matching on the relatively predictable queries and it would remove the need to run Phase twice. What do you think?